File: app/src/main/java/id/ac/polinema/seameo/ecanteen/contract/TransactionContract.java
/*
 * Copyright (c) 2019.
 *
 * Author: Mahatta Maulana
 * Github: https://github.com/hattamaulana
 *
 * Last Modified at 6/21/19 10:45 AM
 */

package id.ac.polinema.seameo.ecanteen.contract;

import com.google.firebase.firestore.QuerySnapshot;

import id.ac.polinema.seameo.ecanteen.model.TransactionModel;
import id.ac.polinema.seameo.ecanteen.presenter.BasePresenter;
import id.ac.polinema.seameo.ecanteen.view.BaseView;

public class TransactionContract {
    public interface Presenter extends BasePresenter {
        void all(Callback callback);
        void get(ItemContract.ScannerResult.Callback callback);
        void save(TransactionModel ob);
    }

    public interface Callback {
        void run(QuerySnapshot q);
    }

    public interface View extends BaseView { }
}

File: app/src/main/java/id/ac/polinema/seameo/ecanteen/contract/TransactionContract.kt
/*
 * Copyright (c) 2019.
 *
 * Author: Mahatta Maulana
 * Github: https://github.com/hattamaulana
 *
 * Last Modified at 6/21/19 10:45 AM
 */

package id.ac.polinema.seameo.ecanteen.contract

import com.google.firebase.firestore.QuerySnapshot

import id.ac.polinema.seameo.ecanteen.model.TransactionModel
import id.ac.polinema.seameo.ecanteen.presenter.BasePresenter
import id.ac.polinema.seameo.ecanteen.view.BaseView

class TransactionContract {
    interface Presenter : BasePresenter {
        fun all(callback: Callback)
        operator fun get(callback: ItemContract.ScannerResult.Callback)
        fun save(ob: TransactionModel)
    }

    interface Callback {
        fun run(q: QuerySnapshot)
    }

    interface View : BaseView
}
